docs/apache-airflow/tutorial.rst

```diff
@@ -376,3 +376,133 @@ Here's a few things you might want to do next:
+Follow the instructions properly to set up Airflow.
+
+Create a Employee table in postgres using this
+
+.. code-block:: sql
+
+  create table "Employees"
+  (
+      "Serial Number" numeric not null
+   constraint employees\_pk
+              primary key,
+      "Company Name" text,
+      "Employee Markme" text,
+      "Description" text,
+      "Leave" integer
+  );
+
+  create table "Employees_temp"
+  (
+      "Serial Number" numeric not null
+   constraint employees\_pk
+              primary key,
+      "Company Name" text,
+      "Employee Markme" text,
+      "Description" text,
+      "Leave" integer
+  );
+
+I have broken this down into 3 steps get data, insert data, merge data.Lets look at the code:
+
+.. code-block:: python
+
+  def get_data():
+      url = "https://docs.google.com/uc?export=download&id=1a0RGUW2oYxyhIQYuezG_u8cxgUaAQtZw"
+
+      payload = {}
+      headers = {}
+
+      response = requests.request("GET", url, headers=headers, data=payload)
+
+      with open('/usr/local/airflow/dags/files/employees.csv', 'w') as file:
+          for row in response.text.split('\n'):
+              file.write(row)
+
+Here we are passing a get request to get the data from the url and save it employees.csv file on our airflow instance
+
+.. code-block:: python
+
+  def insert_data():
+      engine = create_engine("postgresql+psycopg2://postgres:password@localhost:5432/postgres")
+      df = pd.read_csv('/usr/local/airflow/dags/files/employees.csv')
+      df.to_sql('Employees_temp', con=engine, if_exists='replace')
+
+Here we are dumping the file into a temporary table before merging the data to the final employees table
+
+.. code-block:: python
+
+  def merge_data():
+      query = """
+          delete
+          from "Employees" e using "Employees_temp" et
+          where e."Serial Number" = et."Serial Number";
+
+          insert into "Employees"
+          select *
+          from "Employees_temp";
+      """
+      try:
+          engine = create_engine("postgresql+psycopg2://postgres:password@localhost:5432/postgres")
+          session = sessionmaker(bind=engine)()
+          session.execute(query)
+          session.commit()
+          return 0
+      except Exception as e:
+          return 1
+
+Here we are first looking for duplicate values and removing them before we insert new values in our final table
+
+
+Lets look at our DAG:
+
+.. code-block:: python
+
+  with DAG(
+          dag_id='AirflowExample',
+          schedule_interval='0 0 * * *',
+          start_date=datetime.today() - timedelta(days=2),
+          dagrun_timeout=timedelta(minutes=60)
+  ) as dag:
+      t1 = PythonOperator(
+          task_id='get_data',
+          python_callable=get_data,
+      )
+
+      t2 = PythonOperator(
+          task_id='insert_data',
+          python_callable=get_data,
+      )
+
+      t3 = PythonOperator(
+          task_id='merge_data',
+          python_callable=get_data,
+      )
```

Comment:
why not using `@task` decorator?

Reply:
I didn't understand this, can you please elaborate